biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -2282,6 +2304,19 @@ subroutine fuse_patches( csite, bc_in )
     primary_land_fraction_afterfusion = 0._r8
 
     nopatches(1:n_anthro_disturbance_categories) = 0
+
```

Comment:
So for the case when `(hlm_use_nocomp.eq.itrue .and. hlm_use_fixed_biogeog .eq. ifalse)`, each PFT is allotted equal space on all gridcells.  So the number of PFTs on a site is the same as the number of PFTs.  For the case where `(hlm_use_nocomp.eq.itrue .and. hlm_use_fixed_biogeog .eq. itrue)`, then what you suggest here is the case, but at least under the global conditions that I've been running on, the number of PFTs on a gridcell is everywhere less than `maxpatches(primaryforest)`.  Its possible we should put in some more logic up front to handle more possible edge cases, but here I tried to limit the scope of changes in FatesInterfaceMod.

Reply:
actually I think if i use `sum(site_in%use_this_pft)` instead, then this will handle the cases for either value of `hlm_use_fixed_biogeog`. will update.

Reply:
ok I changed the logic.  this is more general and I checked that it works.

Reply:
sounds good.  We really need to enable FATES to dictate maximum number of patches per site, for nocomp, it would be nice to specify this value as some multiple of the number of PFTs.